plugin/pkg/scheduler/scheduler.go

```diff
@@ -90,39 +90,40 @@ func (s *Scheduler) scheduleOne() {
 	dest, err := s.config.Algorithm.Schedule(pod, s.config.NodeLister)
 	if err != nil {
 		glog.V(1).Infof("Failed to schedule: %+v", pod)
-		s.config.Recorder.Eventf(pod, api.EventTypeWarning, "FailedScheduling", "%v", err)
 		s.config.Error(pod, err)
+		s.config.Recorder.Eventf(pod, api.EventTypeWarning, "FailedScheduling", "%v", err)
 		return
 	}
 	metrics.SchedulingAlgorithmLatency.Observe(metrics.SinceInMicroseconds(start))
 
-	b := &api.Binding{
-		ObjectMeta: api.ObjectMeta{Namespace: pod.Namespace, Name: pod.Name},
-		Target: api.ObjectReference{
-			Kind: "Node",
-			Name: dest,
-		},
-	}
+	// Assume optimistically that the binding will succeed and send it
+	// in the background.
+	// The internal cache only assumes a pod anyway and waits for it to be
```

Comment:
So the internal cache has a set of pods that scheduler already bound (i.e. sent binding for them), but it wasn't read via watch yet. In this case, it optimistically assumes that the binding succeeded when binding next pods. However, it is "confirmed" in cache only when read via watch (if it won't be read within some timeout (30s IIRC)), it will be dropped from cache, and will be ignored when binding next pods.

In the currently existing code, the possibility that the pod will not be read via watch after the binding succeeded is very low. On the few workflow, it is higher, because potentially the "binding request" to apiserver may fail. However, this is pretty rare, and even if that happens we are risking only the fact that  we will be assuming that the pod is scheduled for 30s, whereas in fact it isn't. I think the risk here is very low, and it will self-repair itself after this timeout anyway.

I will try to rewrite this comment.